src/compiler/mod.rs

```diff
@@ -2138,8 +2138,8 @@ fn compile_ast_function_call(
             Some(ast::Type::String) => Type::String,
             Some(ast::Type::PublicKey) => Type::PublicKey,
             Some(ast::Type::Bytes) => Type::Bytes,
-            Some(ast::Type::ForeignRecord { collection }) => Type::CollectionReference {
-                collection: collection.clone(),
+            Some(ast::Type::ForeignRecord { contract }) => Type::CollectionReference {
```

Comment:
With [#[serde(rename = ...)]](https://serde.rs/container-attrs.html) we could rename the other identifiers as well.
Although I'm not sure if we even need to support backwards compatibility, we have an `old-main-before-miden-compiler-merge` branch we could use for polybase-rust and if needed, selectively backport some changes to it in the future. @calummoore thoughts?

Reply:
@mateuszmlc  Just to be on the same page, you mean something like so:

```rust
Some(ast::Type::ForeignRecord { contract }) => Type::ContractReference {
        contract: contract.clone(),
    },
```

with `Type` being modified like so:

```rust
#[derive(Clone, Debug, Default, PartialEq, Serialize, Deserialize)]
pub enum Type {
    ...

    #[serde(rename = "CollectionReference")]
    ContractReference {
        #[serde(rename = "collection")]
        contract: String,
    },
    ...
}

```

Is that right?

Reply:
Yes, exactly

Reply:
Implemented and then removed the `rename` in the latest commit (since we're not looking at backward compatibility right now).